src/debugfile.c

```diff
@@ -105,31 +105,35 @@ int debugfile_init (hashcat_ctx_t *hashcat_ctx)
 
   debugfile_ctx->filename = user_options->debug_file;
 
+  HCFILE fp;
+
   if (debugfile_ctx->filename)
   {
-    FILE *fp = fopen (debugfile_ctx->filename, "ab");
-
-    if (fp == NULL)
+    if (hc_fopen (&fp, debugfile_ctx->filename, "ab", HCFILE_FORMAT_PLAIN) == false)
     {
       event_log_error (hashcat_ctx, "Could not open --debug-file file for writing.");
 
       return -1;
     }
 
-    if (lock_file (fp) == -1)
+    if (hc_lockfile (&fp) == -1)
     {
-      fclose (fp);
+      hc_fclose (&fp);
 
       event_log_error (hashcat_ctx, "%s: %s", debugfile_ctx->filename, strerror (errno));
 
       return -1;
     }
 
-    debugfile_ctx->fp = fp;
+    debugfile_ctx->fp = &fp;
   }
   else
   {
-    debugfile_ctx->fp = stdout;
+    fp.is_gzip = false;
+    fp.pfp = stdout;
```

Comment:
What is .pfp variable?

Reply:
because of "fp.fp" is ugly and unreadable ...
.pfp is the FILE * pointer (plain)
.gfp is the gzFile pointer (zlib)